my-project/src/Controller/ProductCategoryCSVController.php

```diff
@@ -71,4 +72,71 @@ public function exportCSV(ProductCategory $productCategory, SerializerService $s
             $dataJson
             ,200);
     }
+
+    /**
+     * @Route("/", name="product_category_import_csv_api", methods="POST")
+     */
+    public function importFromCSV(Request $request, SerializerInterface $serializer, ValidatorInterface $validator,
+                                  ProductCategoryService $productCategoryService)
+    {
+        /*** @var ProductCategory[] $productCategoriesArray */
+        $productCategoriesArray = $serializer->decode(file_get_contents($request->get('csvFile')), 'csv');
+        $numberOfImportedCategories = 0;
+        if(count($productCategoriesArray) == 0)
+        {
+            return new JsonResponse([
+                'error' => 'CSV file is empty',
+                'numberOfImportedCategories' => $numberOfImportedCategories
+            ], 500);
+        }
+
+        if($this->_isOnlyOneRowOfData($productCategoriesArray))
+        {
+            $productCategoriesArray = [$productCategoriesArray];
+        }
+        $em = $this->getDoctrine()->getManager();
+
+        /*** @var ProductCategory[] $tmpProductsCategories */
+        $tmpProductsCategories = [];
+        foreach ($productCategoriesArray as $productCategory) {
+            $tmpProductCategory = $productCategoryService->createProductCategoryFromArray($productCategory);
+
+            $imgPath = $tmpProductCategory->getMainImage();
+            $tmpProductCategory->setMainImage(new File("uploads/images/" . $imgPath));
+            if (count($validator->validate($tmpProductCategory)) != 0) {
+                return new JsonResponse([
+                    'error' => 'Cannot import data from csv. Category '.$productCategory->getName().' not valid.',
+                    'numberOfImportedCategories' => $numberOfImportedCategories
+                ], 500);
+            }
+            $tmpProductCategory->setMainImage($imgPath);
+            $tmpProductsCategories[] = $tmpProductCategory;
+        }
+        foreach ($tmpProductsCategories as $productCategory)
+        {
+            try
+            {
+                $em->persist($productCategory);
+                $em->flush();
+                $numberOfImportedCategories++;
+            }catch(\Exception $exception)
+            {
+                return new JsonResponse([
```

Comment:
Maybe lets not throw exception, lets just log problematic category name with information and continue import
Number of imported categories not needed here